File: scripts/gh_admin_mod/env.py
import os


def get_env(name: str, default: str = "") -> str:
    return os.environ.get(name, default)


def parse_bool(value: str) -> bool:
    return str(value).strip().lower() == "true"


def parse_int(value: str, default: int) -> int:
    try:
        return int(str(value).strip())
    except (TypeError, ValueError):
        return default


def parse_csv(value: str) -> list[str]:
    return [item.strip() for item in str(value).split(",") if item.strip()]


def parse_upper_set(value: str) -> set[str]:
    return {item.upper() for item in parse_csv(value)}


def normalize_user(value: str) -> str:
    return value.strip().lstrip("@").lower()

File: src/env.ts
import * as core from '@actions/core';

export function getEnv(name: string, defaultValue: string = ''): string {
  return process.env[name] || defaultValue;
}

export function parseBool(value: string | boolean | undefined): boolean {
  if (typeof value === 'boolean') return value;
  if (!value) return false;
  return value.trim().toLowerCase() === 'true';
}

export function parseIntSafe(value: string | undefined, defaultValue: number): number {
  if (!value) return defaultValue;
  const parsed = parseInt(value.trim(), 10);
  return isNaN(parsed) ? defaultValue : parsed;
}

export function parseCsv(value: string | undefined): string[] {
  if (!value) return [];
  return value
    .split(',')
    .map((item) => item.trim())
    .filter((item) => item.length > 0);
}

export function parseUpperSet(value: string | undefined): Set<string> {
  return new Set(parseCsv(value).map((item) => item.toUpperCase()));
}

export function normalizeUser(value: string): string {
  return value.trim().replace(/^@/, '').toLowerCase();
}
